target/cheri-common/op_helper_cheri_common.c

```diff
@@ -1109,15 +1218,30 @@ cheri_tag_prot_clear_or_trap(CPUArchState *env, target_ulong va,
             " due to missing CAP_PERM_LOAD_CAP\n", va);
         return 0;
     }
-    if (tag && (prot & PAGE_LC_TRAP))
+    if ((tag && (prot & PAGE_LC_TRAP)) || (prot & PAGE_LC_TRAP_ANY))
         raise_load_tag_exception(env, va, cb, retpc);
     return tag;
 }
 
-bool load_cap_from_memory_raw(CPUArchState *env, target_ulong *pesbt,
-                              target_ulong *cursor, uint32_t cb,
-                              const cap_register_t *source, target_ulong vaddr,
-                              target_ulong retpc, hwaddr *physaddr)
+void squash_mutable_permissions(CPUArchState *env, target_ulong *pesbt,
+                                const cap_register_t *source)
+{
+#ifdef TARGET_AARCH64
+    if (!cap_has_perms(source, CC128_PERM_MUTABLE_LOAD) &&
+        (CAP_cc(cap_pesbt_extract_otype)(*pesbt) == CAP_OTYPE_UNSEALED)) {
+        qemu_maybe_log_instr_extra(env,
+                                   "Squashing mutable load related perms\n");
+        *pesbt &= ~cc128_cap_pesbt_encode_perms(
+            CC128_PERM_MUTABLE_LOAD | CC128_PERM_STORE_LOCAL |
```

Comment:
Though there are loads of these scattered throughout TARGET_AARCH64 code that will need changing to CAP_

Reply:
Also will get rid.